### PR TITLE
arch: arm: nordic: nRF5 to nRF rename for upcoming chip families

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

**Comment by carlescufi** [comment]
CC @gbakke @stig-bjorlykke @nilsMontenegro

File: arch/arm/soc/nordic_nrf/include/nrf_common.h [hunk 1]
```diff
@@ -62,29 +62,29 @@
...
-#define NRF5_UART_BAUDRATE_38400               0x009d0000
-#define NRF5_UART_BAUDRATE_57600               0x00eb0000
-#define NRF5_UART_BAUDRATE_76800               0x013a9000
-#define NRF5_UART_BAUDRATE_115200              0x01d60000
-#define NRF5_UART_BAUDRATE_230400              0x03b00000
-#define NRF5_UART_BAUDRATE_250000              0x04000000
-#define NRF5_UART_BAUDRATE_460800              0x07400000
-#define NRF5_UART_BAUDRATE_921600              0x0f000000
-#define NRF5_UART_BAUDRATE_1000000             0x10000000
+#define NRF_UART_BAUDRATE_300                  0x00014000
```
Comment (ioannisg) [inline arch/arm/soc/nordic_nrf/include/nrf_common.h]: Are these baud-rate values "generic", i.e. not specific to nRF5x SOCs? If so, we need to update the comment in line 65

File: arch/arm/soc/nordic_nrf/include/nrf_common.h [hunk 2]
```diff
@@ -62,29 +62,29 @@
 #define NRF52_IRQ_FPU_IRQn                     38
 
 /**
- * @file UART baudrate divisors for NRF51/NRF52 family processors.
+ * @file UART baudrate divisors for nRF5x family processors.
  *
  * Based on Nordic MDK included header file: nrf52_bitfields.h
```
Comment (ioannisg) [inline arch/arm/soc/nordic_nrf/include/nrf_common.h]: This comment needs an improvement. If we are listing the divisors that are applicable for nRFx SOCs, i.e. nRF51 and nRF52 we need to add here the nrf51_bitfields.h, for consistency.

**Review by ioannisg — CHANGES_REQUESTED** [review]

File: arch/arm/soc/nordic_nrf/include/nrf_common.h (hunk 1 above)
Comment (carlescufi) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: They are generic to Nordic ICs

File: arch/arm/soc/nordic_nrf/include/nrf_common.h (hunk 2 above)
Comment (carlescufi) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: The values in nrf51_bitfields.h were less accurate than the ones for nRF52, that's why we settled on those.

File: arch/arm/soc/nordic_nrf/include/nrf_common.h (hunk 1 above)
Comment (carlescufi) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: Right, I will fix that

Comment (carlescufi) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: Fixed

Comment (ioannisg) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: I wonder: wouldn't it be better to use the MDK headers, when defining the BAUDRATEs? E.g.:
#define NRF_UART_BAUDRATE_9600                ((UARTE_BAUDRATE_BAUDRATE_Baud9600   << UARTE_BAUDRATE_BAUDRATE_Pos) & UARTE_BAUDRATE_BAUDRATE_Msk)

Comment (carlescufi) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: maybe, but that would be outside of the scope of this PR

Comment (ioannisg) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: I can agree with that.

**Comment by ioannisg** [comment]
Looks ok.

**Review by ioannisg — APPROVED** [review]

File: arch/arm/soc/nordic_nrf/include/nrf_common.h (hunk 1 above)
Comment (galak) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: While not related, seems like these defines should move to be drivers/serial/uart_nrf5.c

**Review by galak — COMMENTED** [review]
Looks good, seems like one merge conflict to resolve.

File: arch/arm/soc/nordic_nrf/include/nrf_common.h (hunk 1 above)
Comment (carlescufi) [reply arch/arm/soc/nordic_nrf/include/nrf_common.h]: Yes, I agree. In fact I think they will disappear completely when we switch the serial driver to use our `nrfx` HAL.

**Pull request merged**